crates/cli/src/subcommands/generate/csharp.rs

```diff
@@ -832,25 +827,30 @@ fn autogen_csharp_access_funcs_for_struct(
 ) -> bool {
     let primary_col_idx = schema.pk();
 
-    writeln!(
-        output,
-        "public static System.Collections.Generic.IEnumerable<{struct_name_pascal_case}> Iter()"
-    );
+    writeln!(output, "public static IEnumerable<{struct_name_pascal_case}> Iter()");
     indented_block(output, |output| {
         writeln!(
             output,
-            "foreach(var entry in SpacetimeDBClient.clientDB.GetEntries(\"{table_name}\"))",
+            "return SpacetimeDBClient.clientDB.GetObjects(\"{table_name}\").Cast<{struct_name_pascal_case}>();",
         );
-        indented_block(output, |output| {
-            // TODO: best way to handle this?
-            writeln!(output, "yield return ({struct_name_pascal_case})entry.Item2;");
-        });
     });
+    writeln!(output);
+
+    // Simple alias for Iter().Where(...) for API parity with C# server-side modules.
+    writeln!(
+        output,
+        "public static IEnumerable<{struct_name_pascal_case}> Query(Func<{struct_name_pascal_case}, bool> filter)"
+    );
+    indented_block(output, |output| {
+        writeln!(output, "return Iter().Where(filter);",);
```

Comment:
You need to `include System.Linq;` at the top of the file otherwise the automatically generated files no longer compile.

Reply:
Hm good point, but I'm surprised CI on https://github.com/clockworklabs/spacetimedb-csharp-sdk/pull/93 didn't catch it, as it does build the example...

Reply:
A-ha, it's because the sample project uses `<ImplicitUsings>enable</ImplicitUsings>` in its `csproj` so C# adds `System.Linq` among other usings automatically. I'm going to change that to make those issues easier to catch.